x/emissions/abci.go

```diff
@@ -12,28 +13,47 @@ import (
 
 func BeginBlocker(ctx sdk.Context, keeper keeper.Keeper) {
 
-	reservesFactor, bondFactor, durationFactor := keeper.GetBlockRewardComponents(ctx)
-	blockRewards := reservesFactor.Mul(bondFactor).Mul(durationFactor)
+	emissonPoolBalance := keeper.GetReservesFactor(ctx)
+	if emissonPoolBalance.IsZero() {
+		return
+	}
+	blockRewards, err := keeper.GetFixedBlockRewards()
+	if err != nil {
+		ctx.Logger().Error(fmt.Sprintf("Error while getting fixed block rewards %s", err))
+		return
+	}
 	if blockRewards.IsZero() {
+		ctx.Logger().Error("Block rewards are zero")
 		return
 	}
+	if blockRewards.GT(emissonPoolBalance) {
+		ctx.Logger().Error(fmt.Sprintf("Block rewards %s are greater than emission pool balance %s", blockRewards.String(), emissonPoolBalance.String()))
```

Comment:
Same, I think it should be a warn

Reply:
27c727ade9f75113e34bc452b06286385f41948c